XPCBackupService/Services/BackupUploadService.swift

```diff
@@ -0,0 +1,207 @@
+            childProgress.completedUnitCount = 1
+            throw error
+        }
+    }
+
+    func syncNodeFile(node: BackupTreeNode) async throws -> Int {
+        self.logger.info("Creating file")
+        let childProgress = Progress()
+        backupProgress.addChild(childProgress, withPendingUnitCount: 1)
+
+        var encryptedContentURL: URL? = nil
+        do {
+            encryptedContentURL = try self.getEncryptedContentURL(node: node)
+            guard let safeEncryptedContentURL = encryptedContentURL else {
+                throw BackupUploadError.CannotCreateEncryptedContentURL
+            }
+
+            guard let fileURL = node.url, let inputStream = InputStream(url: fileURL) else {
+                throw BackupUploadError.CannotOpenInputStream
+            }
+
+            let filename = (node.name as NSString)
+            self.logger.info("Starting backing up file \(filename)")
+
+            guard let remoteParentId = node.remoteParentId else {
+                throw BackupUploadError.MissingParentFolder
+            }
+
+            self.logger.info("Remote parent id \(remoteParentId)")
+
+            let result = try await networkFacade.uploadFile(
+                input: inputStream,
+                encryptedOutput: safeEncryptedContentURL,
+                fileSize: Int(fileURL.fileSize),
+                bucketId: self.bucketId
+            ) { completedProgress in
+                childProgress.completedUnitCount = Int64(completedProgress)
+            }
+
+            self.logger.info("Upload completed with id \(result.id)")
+
+            let stringRemoteParentId = "\(remoteParentId)"
+
+            let encryptedFilename = try encrypt.encrypt(
+                string: filename.deletingPathExtension,
+                password: "\(config.CRYPTO_SECRET2)-\(stringRemoteParentId)",
+                salt: cryptoUtils.hexStringToBytes(config.MAGIC_SALT_HEX),
+                iv: Data(cryptoUtils.hexStringToBytes(config.MAGIC_IV_HEX))
+            )
+
+            let createdFile = try await backupAPI.createBackupFile(
+                createFileData: CreateFileData(
+                    fileId: result.id,
+                    type: filename.pathExtension,
+                    bucket: result.bucket,
+                    size: result.size,
+                    folderId: remoteParentId,
+                    name: encryptedFilename.base64EncodedString(),
+                    plainName: filename.deletingPathExtension
+                ),
+                debug: true
+            )
+
+            self.logger.info("✅ Created file correctly with identifier \(createdFile.id)")
+
+            childProgress.completedUnitCount = 1
+
+            if encryptedContentURL != nil {
+                try FileManager.default.removeItem(at: encryptedContentURL!)
+            }
+
+            return createdFile.id
+        } catch {
+            error.reportToSentry()
```

Comment:
If you are rethrowing the error here, don't report the error here, report it where `BackupIUploadService.syncNodeFile()` is being called aka the consumer of this service. Otherwise we are going to report the error possibly twice to Sentry